### DISCUSSION TITLE
Upgrade Help Needed

### DISCUSSION THREAD

**Discussion by Jason-42** [MCLD/greatreadingadventure#749] [Help!]
Hey everyone, I have tried to upgrade from 4.1.1 to 4.2.0 and am having trouble with the automatic database upgrade.  I assumed I did something wrong so I tried again but had the same SQL errors show up in the log.  I am using Server 2016, IIS and SQL 2014 Express.  The log file shows SQL table and Foreign Key issues.  4.1.1 is running fine and I am following the upgrade instructions in the 4.2 install doc.  Anyone have any thought in what I am doing wrong?  

[log-20210422.txt](https://github.com/MCLD/greatreadingadventure/files/6365512/log-20210422.txt)

**Comment by k7hpn** [comment]
Hi Jason. I'm so sorry you are having this issue. This is a bug in the database migrations for adding internationalization to the software.

Possible mitigations for the moment are any one of the following:

- Running the following SQL Query (using SQL Server Management Studio or a similar tool) against the database inserts a default language so that the migration can proceed successfully:

   ```tsql
   IF(NOT EXISTS(SELECT 1 FROM [Languages]))
   BEGIN 
    INSERT INTO [Languages] ([CreatedBy], [CreatedAt], [Description], [IsActive], [IsDefault], [Name]) 
    VALUES (1, GETDATE(), 'English (United States)', 1, 1, 'en-US')
   END
   ```

- Remove all pages and re-insert them after the migration occurs.

- Upgrade to the next release where I'll patch the migration to include this query.

I've opened bug #750 to reflect this and I have a code fix that I'll be putting into the migration to fix this issue for the next release. I was hoping to do a release for the end of this month. Let me know if you want me to send you a fix sooner and I'll see if I can come up with something.

Again: I apologize for your having to deal with this!

**Reply by Jason-42** [reply]
No worries,  I appreciate you responding so quickly. I rolled my VM back to 4.1.1.  I will try the SQL query.  Should I run that query before I do the upgrade to 4.2 or after the upgrade?

**Reply by k7hpn** [reply]
The process I would follow is:

1. Upgrade to 4.2.0
2. Run the application, let that error happen
3. Run the SQL query
4. Restart the app (editing the configuration file would probably cause it to restart, alternately `iisreset` if you are in a Windows/IIS environment)

After that the migration should complete successfully and everything should be good 🤞. Let me know if it's not!

**Reply by Jason-42** [reply]
Thank you.  I will try that and let you know.
